# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/via-cputemp-cha.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

diff --git a/arch/x86/include/asm/cpu_device_id.h b/arch/x86/include/asm/cpu_device_id.h
index 6be777a06944..03b0c6d0760d 100644
--- a/arch/x86/include/asm/cpu_device_id.h
+++ b/arch/x86/include/asm/cpu_device_id.h
@@ -52,6 +52,7 @@
 #define X86_CENTAUR_FAM6_C7_A		0xa
 #define X86_CENTAUR_FAM6_C7_D		0xd
 #define X86_CENTAUR_FAM6_NANO		0xf
+#define X86_CENTAUR_FAM6_CHA		0x47
 
 /* x86_cpu_id::flags */
 #define X86_CPU_ID_FLAG_ENTRY_VALID	BIT(0)
diff --git a/drivers/hwmon/via-cputemp.c b/drivers/hwmon/via-cputemp.c
index 823bff2871e1..81373f9e7b32 100644
--- a/drivers/hwmon/via-cputemp.c
+++ b/drivers/hwmon/via-cputemp.c
@@ -126,15 +126,13 @@ static int via_cputemp_probe(struct platform_device *pdev)
 		data->msr_temp = 0x1423;
 	} else {
 		switch (c->x86_model) {
-		case 0xA:
-			/* C7 A */
-		case 0xD:
-			/* C7 D */
+		case X86_CENTAUR_FAM6_C7_A:
+		case X86_CENTAUR_FAM6_C7_D:
 			data->msr_temp = 0x1169;
 			data->msr_vid = 0x198;
 			break;
-		case 0xF:
-			/* Nano */
+		case X86_CENTAUR_FAM6_NANO:
+		case X86_CENTAUR_FAM6_CHA:
 			data->msr_temp = 0x1423;
 			break;
 		default:
@@ -272,6 +270,7 @@ static const struct x86_cpu_id __initconst cputemp_ids[] = {
 	X86_MATCH_VENDOR_FAM_MODEL(CENTAUR, 6, X86_CENTAUR_FAM6_C7_A,	NULL),
 	X86_MATCH_VENDOR_FAM_MODEL(CENTAUR, 6, X86_CENTAUR_FAM6_C7_D,	NULL),
 	X86_MATCH_VENDOR_FAM_MODEL(CENTAUR, 6, X86_CENTAUR_FAM6_NANO,	NULL),
+	X86_MATCH_VENDOR_FAM_MODEL(CENTAUR, 6, X86_CENTAUR_FAM6_CHA,	NULL),
 	X86_MATCH_VENDOR_FAM_MODEL(CENTAUR, 7, X86_MODEL_ANY,		NULL),
 	{}
 };
